File: API/BetaCycleAPI/BetaCycleAPI/Models/AddressFormData.cs
namespace BetaCycleAPI.Models
{
    public class AddressFormData
    {
        /// <summary>
        /// First street address line.
        /// </summary>
        public string AddressLine1 { get; set; }

        /// <summary>
        /// Second street address line.
        /// </summary>
        public string? AddressLine2 { get; set; }

        /// <summary>
        /// Name of the city.
        /// </summary>
        public string City { get; set; }

        /// <summary>
        /// Name of state or province.
        /// </summary>
        public string StateProvince { get; set; }

        public string CountryRegion { get; set; }

        /// <summary>
        /// Postal code for the street address.
        /// </summary>
        public string PostalCode { get; set; }

        /// <summary>
        /// The kind of Address. One of: Archive, Billing, Home, Main Office, Primary, Shipping
        /// </summary>
        public string AddressType { get; set; }

        /// <summary>
        /// Date and time the record was last updated.
        /// </summary>
        public DateTime ModifiedDate { get; set; }
    }
}

File: FrontEnd/src/app/shared/models/AddressFormData.ts
import { Address } from "./Address";
import { CustomerAddress } from "./CustomerAddress";

export class AddressFormData {

   addressLine1: string;
   addressLine2: string;
   city: string;
   stateProvince: string;
   countryRegion: string;
   postalCode: string;
   addressType: string;
   modifiedDate: Date = new Date(Date.now());
   isDeleted: boolean = false;
  
   constructor (
    addressLine1: string,
    addressLine2: string = '',
    city: string,
    stateProvince: string,
    countryRegion: string,
    postalCode: string,
    addressType: string,
    isDeleted: boolean
   ) {
    this.addressLine1 = addressLine1;
    this.addressLine2 = addressLine2;
    this.city = city;
    this.stateProvince = stateProvince;
    this.countryRegion = countryRegion;
    this.postalCode = postalCode;
    this.addressType = addressType;
    this.isDeleted = isDeleted;
   }

}